src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -0,0 +1,198 @@
+          </xsd:element>
+          <xsd:element name="data">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+                <xsd:element name="comment" type="xsd:string" minOccurs="0" msdata:Ordinal="2" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" msdata:Ordinal="1" />
+              <xsd:attribute name="type" type="xsd:string" msdata:Ordinal="3" />
+              <xsd:attribute name="mimetype" type="xsd:string" msdata:Ordinal="4" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="resheader">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" />
+            </xsd:complexType>
+          </xsd:element>
+        </xsd:choice>
+      </xsd:complexType>
+    </xsd:element>
+  </xsd:schema>
+  <resheader name="resmimetype">
+    <value>text/microsoft-resx</value>
+  </resheader>
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="ArrayDepthTooLarge" xml:space="preserve">
+    <value>Depth of {0} within an array is larger than max depth of {1}.</value>
+  </data>
+  <data name="ArrayEndWithinObject" xml:space="preserve">
+    <value>We are within an object but observed an ']'.</value>
+  </data>
+  <data name="Default" xml:space="preserve">
+    <value>An error occurred while reading the json.</value>
+  </data>
+  <data name="DepthMustBePositive" xml:space="preserve">
+    <value>Mismatched number of start/end objects or arrays. Depth is {0} but must be greater than 0.</value>
+  </data>
+  <data name="EndOfCommentNotFound" xml:space="preserve">
+    <value>Expected either a '\n' or '*/' to indicate end of comment, but instead reached end of data.</value>
+  </data>
+  <data name="EndOfStringNotFound" xml:space="preserve">
+    <value>Expected a '"' to indicate end of string, but instead reached end of data.</value>
+  </data>
+  <data name="ExpectedDigitNotFound" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected a digit ('0'-'9').</value>
+  </data>
+  <data name="ExpectedDigitNotFoundEndOfData" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected a digit ('0'-'9') but reached end of data.</value>
```

Comment:
Is it actually important to distinguish the "I ran out of data" cases in exception text?

`"End of data was reached unexpectedly."` : Covers a bare string with no close quote, a bare number ending with a '.' (unless that's allowed), an open object or array with no matching close, etc.